lightstep/util.py

```diff
@@ -20,9 +20,7 @@ def _generate_guid():
     """
     Construct a guid - random 64 bit integer converted to a string.
     """
-    # Note: uuid.uuid4() returns 128 bit int. To get 64 bit int, apply the mask.
-    guid = uuid.uuid4().int & (1<<64)-1
-    return str(guid)
+    return str(random.getrandbits(64))
```

Comment:
BTW I'm referring to python2.7

ᐧ

On Wed, Apr 13, 2016 at 4:17 PM, Josh MacDonald <jmacd@lightstep.com> wrote:

> The default Random() instance uses urandom for a seed, which ought to be
> good. It falls back to a seed derived from time.time() which is probably
> less good.
>
> I looked at basictracer-go and it just uses the timestamp to seed its RNG,
> and it only uses 63 bits of randomness.
> ᐧ
>
> On Wed, Apr 13, 2016 at 4:13 PM, bhs <notifications@github.com> wrote:
>
>> In lightstep/util.py
>> <https://github.com/lightstep/lightstep-tracer-python/pull/1#discussion_r59641756>
>> :
>>
>> > @@ -20,9 +20,7 @@ def _generate_guid():
>> >      """
>> >      Construct a guid - random 64 bit integer converted to a string.
>> >      """
>> > -    # Note: uuid.uuid4() returns 128 bit int. To get 64 bit int, apply the mask.
>> > -    guid = uuid.uuid4().int & (1<<64)-1
>> > -    return str(guid)
>> > +    return str(random.getrandbits(64))
>>
>> perhaps we should seed the rng (or use jumpahead) somewhere... I'm
>> nervous about two processes that don't otherwise use this package getting
>> the same numbers. (I didn't look to see whether random's global RNG is
>> seeded by default)
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/lightstep/lightstep-tracer-python/pull/1/files/2de494810b73dd69c6b4bb87e87007291309d573#r59641756>
>>
>
>